src/CoffeeShop/CoffeeShopsArray.php

```diff
@@ -0,0 +1,24 @@
+<?php
+namespace CoffeeShop;
+
+class CoffeeShopsArray extends \ArrayObject{
+
+    public function __construct($shopList) {
+        $list = $this->filterList($shopList);
+
+        parent::__construct($list);
+    }
+
+    public function filterList($list){
+        $newList = [];
+
+        $filterCB = function($item){
+            $v = ($item instanceof IShop);
```

Comment:
What else can it be in the $list param?

Reply:
In PHP each item from an array can be anything. So, here I'm trying to ensure that this array will contains only shops.

Reply:
Want to comes into my mind right now is to instantiate each item of the array at the creation time.
The constructor get only the csv list as array.